external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -419,6 +416,19 @@ private[kafka010] class KafkaOffsetReader(
     stopConsumer()
     _consumer = null  // will automatically get reinitialized again
   }
+
+  private def getPartitions(): ju.Set[TopicPartition] = {
+    var partitions = Set.empty[TopicPartition].asJava
+    val startTimeMs = System.currentTimeMillis()
+    while (partitions.isEmpty && System.currentTimeMillis() - startTimeMs < pollTimeoutMs) {
```

Comment:
Better to put some sleep in a loop if poll(0) doesn't employ any sleep. If the duration is somewhat big value, exponential sleep would be ideal.

Reply:
+1 on having sleep not to be too aggressive, will check whether exponential sleep is a custom in Spark.

Reply:
I've added sleep in the `poll` as timeout:
* If `poll` succeeds it returns immediately
* If no assignment the it waits until the specified timeout.